Modules/Microsoft365DSC/DSCResources/MSFT_TeamsFederationConfiguration/MSFT_TeamsFederationConfiguration.psm1

```diff
@@ -0,0 +1,279 @@
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    try
+    {
+        $config = Get-CsTenantFederationConfiguration -ErrorAction Stop
+
+        return @{
+            Identity                  = $Identity
+            AllowFederatedUsers       = $config.AllowFederatedUsers
+            AllowPublicUsers          = $config.AllowPublicUsers
+            AllowTeamsConsumer        = $config.AllowTeamsConsumer
+            AllowTeamsConsumerInbound = $config.AllowTeamsConsumerInbound
+            Credential                = $Credential
+        }
+    }
+    catch
+    {
+        try
+        {
+            Write-Verbose -Message $_
+            $tenantIdValue = ""
+            if (-not [System.String]::IsNullOrEmpty($TenantId))
+            {
+                $tenantIdValue = $TenantId
+            }
+            elseif ($null -ne $Credential)
+            {
+                $tenantIdValue = $Credential.UserName.Split('@')[1]
+            }
+            Add-M365DSCEvent -Message $_ -EntryType 'Error' `
+                -EventID 1 -Source $($MyInvocation.MyCommand.Source) `
+                -TenantId $tenantIdValue
+        }
+        catch
+        {
+            Write-Verbose -Message $_
+        }
+        throw $_
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        [ValidateSet('Global')]
+        $Identity,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowFederatedUsers,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowPublicUsers,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowTeamsConsumer,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowTeamsConsumerInbound,
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $Credential
+    )
+
+    Write-Verbose -Message "Setting configuration of Teams Federation"
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftTeams' `
+        -InboundParameters $PSBoundParameters
+
+    $SetParams = $PSBoundParameters
+    $SetParams.Remove("Credential")
+
+    Set-CsTenantFederationConfiguration @SetParams
```

Comment:
Based on the definition of the Set-CsTenantFederationConfiguration (https://docs.microsoft.com/en-us/powershell/module/skype/set-cstenantfederationconfiguration?view=skype-ps) it doesn't accept parameter Identity. Can you validate if this is a bug or a miss in the documentation?

Reply:
The cmdlet accepts the Identity parameter: 
![image](https://user-images.githubusercontent.com/37979749/155132464-a3cf7b57-16d4-4ca7-9f9b-0d431064c514.png)